src/components/RepoCard.astro

```diff
@@ -0,0 +1,157 @@
+---
+import { Icon } from "astro-icon"
+
+export type Props = {
+	title: string
+	sourceUrl: string
+	stackblitzUrl: string
+	codesandboxUrl: string
+	gitpodUrl: string
+	previewImage: string | null
+}
+
+const {
+	title,
+	sourceUrl,
+	stackblitzUrl,
+	codesandboxUrl,
+	gitpodUrl,
+	previewImage,
+} = Astro.props
+---
+
+<article class="relative border border-astro-gray-500 bg-astro-gray-600">
+	<div class="noise z-0"></div>
+	<div class="aspect-video bg-astro-gray-700">
+		{
+			!!previewImage && (
+				<img src={previewImage} alt="" class="object-cover s-full" />
+			)
+		}
+	</div>
+	<div class="flex flex-col gap-4 p-5">
+		<h3 class="heading-4">{title}</h3>
+		<p>
+			<a href={sourceUrl} class="inline-flex items-center gap-1">
+				<Icon name="github" size={16} />
```

Comment:
similar to Chris's comment above, most of the `<Icon>` components could probably use `aria-hidden="true"`

If any of the SVGs really need to be accessible, an inline `<title>` in the SVG file is the best bet